spec/rubocop/cop/style/fetch_env_var_spec.rb

```diff
@@ -336,4 +304,343 @@
+            expect_offense(<<~RUBY)
+              z || ENV['X'] || y || next
+                   ^^^^^^^^ Use `ENV.fetch('X') { y }` instead of `ENV['X'] || y`.
+            RUBY
+
+            expect_correction(<<~RUBY)
+              z || ENV.fetch('X') { y } || next
+            RUBY
+          end
+        end
+      end
+    end
+
+    context 'and followed by multiple `ENV[]`s' do
+      context 'when the `ENV`s are separated by non-ENV nodes' do
+        context 'when the node is the left end of `||` chains' do
+          it 'registers an offense' do
+            expect_offense(<<~RUBY)
+              ENV['X'] || y || ENV['Z'] || a || ENV['B']
+                                                ^^^^^^^^ Use `ENV.fetch('B')` or `ENV.fetch('B', nil)` instead of `ENV['B']`.
+            RUBY
+
+            expect_correction(<<~RUBY)
+              ENV.fetch('X') { y } || ENV.fetch('Z') { a } || ENV.fetch('B', nil)
+            RUBY
+          end
+        end
+
+        context 'when the node is between `||`s' do
+          it 'registers an offense' do
+            expect_offense(<<~RUBY)
+              z || ENV['X'] || y || ENV['Z'] || a || ENV['B']
+                                                     ^^^^^^^^ Use `ENV.fetch('B')` or `ENV.fetch('B', nil)` instead of `ENV['B']`.
+            RUBY
+
+            expect_correction(<<~RUBY)
+              z || ENV.fetch('X') { y } || ENV.fetch('Z') { a } || ENV.fetch('B', nil)
+            RUBY
+          end
+        end
+      end
+
+      context 'when two `ENV[]`s are next to each other' do
+        context 'when the node is the left end of `||` chains' do
+          it 'registers an offense' do
+            expect_offense(<<~RUBY)
+              ENV['X'] || ENV['Y'] || a
+                          ^^^^^^^^ Use `ENV.fetch('Y') { a }` instead of `ENV['Y'] || a`.
+            RUBY
+
+            expect_correction(<<~RUBY)
+              ENV.fetch('X') { ENV.fetch('Y') { a } }
+            RUBY
+          end
+        end
+
+        context 'when the node is between `||`s' do
+          it 'registers an offense' do
+            expect_offense(<<~RUBY)
+              z || ENV['X'] || ENV['Y'] || a
+                               ^^^^^^^^ Use `ENV.fetch('Y') { a }` instead of `ENV['Y'] || a`.
+            RUBY
+
+            expect_correction(<<~RUBY)
+              z || ENV.fetch('X') { ENV.fetch('Y') { a } }
+            RUBY
+          end
+        end
+      end
+    end
+
+    context 'and followed by a mutiline expression' do
+      context 'when the node is the left end of `||` chains' do
+        it 'registers an offense' do
+          expect_offense(<<~RUBY)
+            ENV['X'] || y.map do |a|
+            ^^^^^^^^ Use `ENV.fetch('X')` with a block containing `y.map do |a| ...`
+              a * 2
+            end
+          RUBY
+
+          expect_correction(<<~RUBY)
+            ENV.fetch('X') do
+              y.map do |a|
+                a * 2
+              end
+            end
```

Comment:
Autocorrect doesn't indent properly when `ENV` is indented.

```ruby
# With no indentation
ENV['X'] || y.map do |a|
  a * 2
end

ENV.fetch('X') do
  y.map do |a|
    a * 2
  end
end

# With an indentation
def foo
  ENV['X'] || y.map do |a|
    a * 2
  end
end

def foo
  ENV.fetch('X') do
    y.map do |a|
        a * 2
      end
end
end
```

Reply:
Fixed: https://github.com/rubocop/rubocop/pull/10585/commits/214f6f0b5e5cafa148aa4426e5ecba2a9d504ab9